.github/workflows/diff.yaml

```diff
@@ -75,14 +75,30 @@ jobs:
       - name: Fetch all history for all tags and branches
         run: git fetch
 
-      - name: Build combined ASAN, UBSAN and coverage binaries
+      - name: Initialize deps
         run: |
           # Activate toolchain.
           source /opt/toolchain-v4/activate
 
           # Initialize dependencies.
           ./init
 
+      - name: Changed Files
+        uses: tj-actions/changed-files@v34.3.2
+
+      - name: Python code analysis
+        run: |
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            if [[ $file == *.py ]]; then
+              black $file --check --diff
+            fi
+          done
```

Comment:
Move this into a script, so that it can also be run locally? Maybe include options to analyze and fix the code in-place?

Reply:
I am against this since it is really trivial to run black all one needs to do is `black .`, or have a `pre-commit` hook installed, this is just a CI verification all is correct. This part is only for checking the format of all changed files, locally `steps.changed-files.outputs.all_changed_files` does not exist, and --check and --diff flags are useless locally since they only inform what is wrong. What do you say?